lib/evmone/eof.cpp

```diff
@@ -204,12 +219,12 @@ std::variant<EOFSectionHeaders, EOFValidationError> validate_section_headers(byt
         return EOFValidationError::section_headers_not_terminated;
 
     const auto section_bodies_without_data =
-        section_headers[TYPE_SECTION].front() +
+        static_cast<uint64_t>(section_headers.type_size) +
```

Comment:
```suggestion
        uint64_t{section_headers.type_size} +
```

I'm not sure about uint64_t. Can't we keep it as uint32_t?

Reply:
Note to myself: this accumulation can be replaced with prefix sum to compute offsets in the same time.

Reply:
> I'm not sure about uint64_t. Can't we keep it as uint32_t?

We cannot, actually. In case there's huge section sizes declared, but body is truncated, we might pass the contract limit size (before this function), and get 32-bit overflow here (before body truncation is checked below).

Reply:
I've added a unit test so far.